dex/src/main/scala/com/wavesplatform/dex/api/http/directives/ProtectDirective.scala

```diff
@@ -0,0 +1,34 @@
+package com.wavesplatform.dex.api.http.directives
+
+import akka.http.scaladsl.server._
+import com.wavesplatform.dex.api.http.entities.{InvalidJsonResponse, WavesNodeUnavailable}
+import com.wavesplatform.dex.api.http.{HasStatusBarrier, PlayJsonException}
+import com.wavesplatform.dex.api.routes.ApiRoute
+import com.wavesplatform.dex.error._
+import com.wavesplatform.dex.grpc.integration.exceptions.WavesNodeConnectionLostException
+
+trait ProtectDirective extends ApiRoute with HasStatusBarrier {
```

Comment:
I think we should use the "cake" pattern here because MatcherApiRoute already inherits these members. so it would be sth like

```trait ProtectDirective { self: ApiRoute with HasStatusBarrier =>```